.github/workflows/main-build.yml

```diff
@@ -68,9 +68,37 @@ jobs:
           name: coverage-report
           path: all/build/reports/jacoco/test/html
 
+  markdown-link-check:
+    # release branches are excluded because links to external urls can break at any time which
+    # requires unnecessary release branch maintenance
+    if: ${{ !startsWith(github.ref_name, 'v') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Check markdown links
+        # using retry because of sporadic external site failures
+        uses: nick-invision/retry@v2.6.0
+        with:
+          # timing out has not been a problem
+          timeout_minutes: 15
+          # give external site some time to hopefully recover
+          retry_wait_seconds: 120
+          max_attempts: 5
+          command: |
+            npm install -g markdown-link-check
+            find . -type f \
+                   -name '*.md' \
+                   -not -path './.github/*' \
+                   -not -path './node_modules/*' \
+                   -print0 \
+              | xargs -0 -n1 markdown-link-check --config .github/scripts/markdown_link_check_config.json
+
   publish-snapshots:
     name: Publish snapshots
     if: ${{ github.event_name == 'push' && github.repository == 'open-telemetry/opentelemetry-java' }}
+    # not blocking snapshot publishing on markdown-link-check because links to external urls can
+    # break at any time
```

Comment:
This comment pertains to the `if` online 99, correct? If so, can we place the comment before line 99?